radixdlt/babylon-wallet-android
# Use state repository to query dApps

## Description
* In this pr the `DAppRepository` is deleted and replaced by the single method of `getDAppDetails` in `StateRepository`. That made sense since the dApps are queried from StateApi and have the same structure as fungibles, non fungibles, components etc...

   Also when in need to query dapps with associated resources (claimed entities), we could reuse methods from the repository to combine dapp information with resource information. There was no need for a separate repository.

   In order to accomplish that a new entity in the database was introduced which caches dapps and their metadata.

* Created a specific use case to verify an incoming dapp. This was not needed to be a separate method in the repository.
* Created a specific use case that can be used to verify claimed websites that dapps claim. This is done by querying the first claimed website and crosschecking the dapp definition. This was not done before.
   In order to achieve that I had to also move the dapp sheet into its own dialog (as we do for resources) so as to share this logic in persona details and transaction review. With this way we can create a view model that besides querying the dapp with resources, does the website check. Until the check is completed, a shimerring effect is shown in the website section. I put the same logic in the `DappDetailsViewModel`.
* Last but not least the rest of the screens that would need only dapp information (and not resources) like the Connected Dapps screen, I just used another use case that fetches dapp information and not associated resource info. That way it is a bit faster.

## How to test
1. Connect to different dapps, either with dev mode on or off
2. Click on the connected dapps
3. Click on persona's connected dapps
4. Do some transactions with integrations tests in dashboard. That way you will be able to see the using dapps section and the badges.

## PR submission checklist
